src/agoradatatools/etl/transform/model_details.py

```diff
@@ -0,0 +1,190 @@
+"""
+This module contains the transformation logic for the model_details datasets.
+This is for the Model AD project.
+"""
+import pandas as pd
+from typing import Any, Dict, List
+
+
+def transform_model_details(
```

Comment:
This function is pretty long and has a lot of logic for handling different dataframes all included. Do you think it would be a good idea to break this up with helper functions? Anything that needs to be applied to all dataframes could be grouped together while anything specific could be separated out for each df type. Similar to some of Jaclyns' comments, I would also try to reduce the amount of repeated code here as much as possible too.